### PR TITLE
tplg: add tplg file for CML with da7219 + max98357

### PR DESCRIPTION
**Pull request by macchian**
add the topology file for Cometlake
with analog codec da7219 on SSP0 for headphone playback/headset mic
with digital codec max98357 on SSP1 for speaker playback

Signed-off-by: Mac Chiang mac.chiang@intel.com

### CODE REVIEW

**Comment by macchian** [comment]
[CMakeLists.txt](https://github.com/thesofproject/sof/files/3275686/CMakeLists.txt)

File: tools/topology/CMakeLists.txt [hunk 1]
```diff
@@ -1,71 +1,18 @@
...
-	)
+project(SOF_TOOLS C)
+
+set(SOF_ROOT_SOURCE_DIRECTORY "${PROJECT_SOURCE_DIR}/..")
 
-	add_custom_target(topology_${machine} DEPENDS ${machine}.tplg)
-	add_dependencies(topologies topology_${machine})
-endforeach()
+add_subdirectory(logger)
+add_subdirectory(eqctl)
```
Comment (lgirdwood) [inline tools/topology/CMakeLists.txt]: This is just called ctl now rather than eqctl since the code was generic and could be used with more than just EQs.

Comment (jajanusz) [reply tools/topology/CMakeLists.txt]: @lgirdwood But it's PR for old branch. Not master.

**Comment by jajanusz** [comment]
@macchian Also you are missing Signed-off-by line. We require this for contributions.

**Comment by macchian** [comment]
> @macchian You requested merge to branch that is release branch. Release branch is meant to be frozen, unless you have very good reason to add something there. May I ask you why you want to do it?

I'm not aware of the condition on release branch. Could you recommend the branch I should pull request? the master branch is ok?

**Comment by jajanusz** [comment]
@macchian You can add new topologies with pull request to master branch.

**Comment by ranj063** [comment]
@macchian is there a need for a new m4 file. Can we reuse sof-glk-da7219.m4 if the codec combination is the same?

**Comment by macchian** [comment]
> @macchian is there a need for a new m4 file. Can we reuse sof-glk-da7219.m4 if the codec combination is the same?

I'm afraid not. Because the I2s MCLK/BCLK/LRCLK clock settings is different between glk and cml.

**Comment by ranj063** [comment]
> I'm afraid not. Because the I2s MCLK/BCLK/LRCLK clock settings is different between glk and cml.

@macchian what I meant was to use the same m4 file and set these values based on the platform. Take a look at sof-cml-rt5682.m4 for example. It is the same m4 file for both cml and whl

**Comment by plbossart** [comment]
> > I'm afraid not. Because the I2s MCLK/BCLK/LRCLK clock settings is different between glk and cml.
> 
> @macchian what I meant was to use the same m4 file and set these values based on the platform. Take a look at sof-cml-rt5682.m4 for example. It is the same m4 file for both cml and whl

the clock tree is the same for WHL/CML. It's different for GLK/CML, so you probably need to have two m4 files.

**Comment by jajanusz** [comment]
@macchian Please change target branch. You cannot push to release branch. Probably you want to merge it to master.

**Comment by macchian** [comment]
> @macchian Please change target branch. You cannot push to release branch. Probably you want to merge it to master.

Yes, I changed to mmaster branch. Please review. Thanks.

**Comment by jajanusz** [comment]
@ranj063 @macchian Is it the same as #1662 ? If so, please close this one.

**Pull request closed without merging**